internal/oidc/discovery/discovery_handler.go

```diff
@@ -37,6 +37,7 @@ type Metadata struct {
 	TokenEndpointAuthMethodsSupported []string `json:"token_endpoint_auth_methods_supported"`
 	ScopesSupported                   []string `json:"scopes_supported"`
 	ClaimsSupported                   []string `json:"claims_supported"`
+	CodeChallengeMethosSupport        []string `json:"code_challenge_methods_supported"`
```

Comment:
Could we change the name of this field to `CodeChallengeMethodsSupported`? Seems like maybe this was a typo.

Also, I think this test expectation in the integration tests will need to be updated to include the new field... https://github.com/vmware-tanzu/pinniped/blob/53597bb82492f134c0b4fb01d4173a19922a7457/test/integration/supervisor_discovery_test.go#L499-L512

Reply:
You are right! I fixed the typo, and added the field to the new test.